expression/integration_test.go

```diff
@@ -7516,6 +7516,15 @@ func (s *testIntegrationSuite2) TestIssue15847(c *C) {
 	tk.MustExec("drop view if exists t15847")
 }
 
+func (s *testIntegrationSerialSuite) TestLikeWithCollation(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	collate.SetNewCollationEnabledForTest(true)
+	defer collate.SetNewCollationEnabledForTest(false)
+
+	tk.MustQuery(`select 'a' like 'A' collate utf8mb4_general_ci;`).Check(testkit.Rows("1"))
```

Comment:
Please add the test case mentioned in https://github.com/pingcap/tidb/issues/21892.

Reply:
That test case requires `unicode_ci` support. I will add it after `unicode_ci` has been cherry-picked.

Reply:
Now it is done, please update it. Thanks!

Reply:
need wait  for infer collation PR 😂